engine/src/test/java/org/hibernate/search/test/dsl/NumericTypeWithNullEncodingTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.test.dsl;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.lucene.search.NumericRangeQuery;
+import org.apache.lucene.search.Query;
+import org.hibernate.search.annotations.DocumentId;
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+import org.hibernate.search.backend.spi.Work;
+import org.hibernate.search.backend.spi.WorkType;
+import org.hibernate.search.query.dsl.QueryBuilder;
+import org.hibernate.search.query.engine.spi.EntityInfo;
+import org.hibernate.search.testsupport.TestForIssue;
+import org.hibernate.search.testsupport.junit.SearchFactoryHolder;
+import org.hibernate.search.testsupport.setup.TransactionContextForTest;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * A query targeting a numeric-encoded property needs to be a NumericQuery even if it's using 'indexNullAs' markers.
+ *
+ * @author Sanne Grinovero (C) 2015 Red Hat Inc.
+ */
+@TestForIssue(jiraKey = "HSEARCH-1973")
+public class NumericTypeWithNullEncodingTest {
+
+	@Rule
+	public SearchFactoryHolder sfHolder = new SearchFactoryHolder( SomeEntity.class );
+
+	@Before
+	public void prepareTestData() {
+		storeData( "title-one", 1 );
+		storeData( "title-two", 2 );
+		storeData( "title-three", 3 );
+	}
+
+	@Test
+	public void verifyExplicitRangeQuery() {
+		Query query = getQueryBuilder()
+					.range()
+						.onField( "age" )
+						.from( 1 ).excludeLimit()
+						.to( 3 ).excludeLimit()
+						.createQuery();
+
+		Assert.assertTrue( query instanceof NumericRangeQuery );
+
+		List<EntityInfo> queryEntityInfos = runProjection( query, "title" );
+
+		Assert.assertEquals( 1, queryEntityInfos.size() );
+		EntityInfo entityInfo = queryEntityInfos.get( 0 );
+		Assert.assertEquals( "title-two", entityInfo.getProjection()[0] );
+	}
+
+
+	@Test
+	public void verifyExplicitKeywordQuery() {
+		Query query = getQueryBuilder()
+					.keyword()
+					.onField( "age" )
+					.matching( 2 )
```

Comment:
Nope. I'm just testing this in infinispan and get 
>org.hibernate.search.exception.SearchException: HSEARCH000237: Cannot create numeric range query for field 'age', since from and to values are null
	at org.hibernate.search.bridge.util.impl.NumericFieldUtils.createNumericRangeQuery(NumericFieldUtils.java:43)